resources/default_pipeline_environment.yml

```diff
@@ -232,7 +232,8 @@ steps:
     logSuccessfulMavenTransfers: false
   mtaBuild:
     buildTarget: 'NEO'
-    mtaJarLocation: 'mta.jar'
+    mtaJarLocation: '/opt/sap/mta/lib/mta.jar'
```

Comment:
I would like to suggest to get ride of having the path here.

In the `mta-archive-builder` docker image we have a script which hides the path which is basically an implementation detail. See [here](https://raw.githubusercontent.com/SAP/devops-docker-images/master/mta-archive-builder/scripts/mtaBuild.sh). Should be sufficient to call `mtaBuild.sh`.

Reply:
What about custom images? Do they need to have `mtaBuild.sh`? Can we get rid of `mtaJarLocation` as an option? Not exactly sure how that should work.

Reply:
This will break compatibility in the library, won't it? I think we have to retain that option, and thus keep the default value. The `mtaBuild` script is nice for using the image directly, but I don't see how to use it in the lib without breaking compatibility.

Reply:
> Do they need to have [...]

yes. As part of the contract between piper lib and the image. This is IMO better than making that configurable. When I use an image I do not want to deal with implementation details of the image (e.g. where is the jar file located in the image).

> This will break compatibility in the library,
Yes. I agree we have of course to be careful with incompatible changes. I have no problem if we keep this for some time for compatibility reasons. But in the log run we should not have this property.

The point is: backward-compatibility is not a value "as it". Almost everything in life is not a value "as it", but serves other higher values. Changing code in an incompatible way means: creating integration efforts on the user side. Not changing code in an incompatible way means: creating efforts on our side in terms of backward compatibility layers causing more code maintenance efforts. Now we have to find a balance: We can say: we stay backward compatible for the next 100 years, causing high efforts on our side or we can say zero backward compatibility causing no coresponding efforts on our side but huge efforts for users, which will refuse using the code in this case  most probably.

What we need is a habit of "Maß und Vernunft" (moderate et prudenter) and some good rules for managing backward compatibiltiy. The currently agreed approach for this project is: "announce backward compatible changes in the release notes one release before we do it". Another possible way would be to stick to semantic versioning.

Reply:
Ok, let's just agree that I have not introduced this option in this PR. I'm not sure when I have the capacity for that unplanned work, meanwhile our users need this default configuration. I'd vote for merging this now and and re-working the step later.